mediasms.go

```diff
@@ -0,0 +1,499 @@
+package mediasms
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// Client struct
+type Client struct {
```

Comment:
Do we need an exposed method to instantiate the Client (e.g. `NewClient()`)?
Or does the consumer of the library needs to instantiate the Client themselves?

Reply:
Constructor functions such as `NewClient()` which would I assume return a `Client` struct with values set are generally bad practice in go, are they not?

Reply:
I guess we could make a factory function that just returns a `Client`. Not sure if we want to go that route. I suppose it could guarantee that they have supplied the `username` and `password` and something for `prefix`.